schema/spacewalk/spacewalk-schema-upgrade

```diff
@@ -96,7 +96,7 @@ if ($ENV{SUMA_TEST_SCHEMA_VERSION}) {
 }
 
 my $target_schema = join '-', $my_schema_name, $my_schema_version, $my_schema_release;
-(my $target_schema_norm = $target_schema) =~ s!^(.+-\d+(\.\d+)*)(\..*)*$!$1!;
+(my $target_schema_norm = $target_schema) =~ s!^(.+-\d+(\.\d+)*-[a-z0-9]*?)(\..*)*$!$1!;
```

Comment:
I wonder if both '\*' and '?' are really needed? I would think a '\*' would be enough.
Another point is, that it would not catch upper case A-Z. So if the next one think to use 'LP' we would have the
same problem again.

And the dash (-) is now required, but the release not.
So without having it tested I would suggest to try:
```
(my $target_schema_norm = $target_schema) =~ s!^(.+-\d+(\.\d+)*-?[a-zA-Z0-9]*)(\..*)*$!$1!;
 ```
Would this work as well?

Reply:
I'm ok with all suggestions so far. And it's looking good within the 'regex coach'. But the dash (-) is a requirement in the original code. So I wouldn't change this. It's currently looking for ```n-v-r.%{dist}``` and should strip the ```%{dist}``` if present. So
```
(my $target_schema_norm = $target_schema) =~ s!^(.+-\d+(\.\d+)*-[a-zA-Z0-9]*)(\..*)*$!$1!;
```
should be good.

Reply:
ok, would be ok, for me.